lxd/patches.go

```diff
@@ -2504,6 +2505,187 @@ func patchStorageApiCephSizeRemove(name string, d *Daemon) error {
 	return nil
 }
 
+func patchDevicesNewNamingScheme(name string, d *Daemon) error {
+	cRegular, err := d.db.ContainersList(db.CTypeRegular)
+	if err != nil {
+		logger.Errorf("Failed to retrieve containers from database")
+		return err
+	}
+
+	cSnapshots, err := d.db.ContainersList(db.CTypeSnapshot)
+	if err != nil {
+		logger.Errorf("Failed to retrieve snapshots from database")
+		return err
+	}
+
+	cts := append(cRegular, cSnapshots...)
+	if len(cts) == 0 {
+		logger.Debugf("No containers found")
+		return nil
+	}
+
+	for _, ct := range cts {
+		devicesPath := shared.VarPath("devices", ct)
+		devDir, err := os.Open(devicesPath)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				logger.Debugf("Container \"%s\" does not have on-disk devices", ct)
+				return err
+			}
```

Comment:
A container which has never started will not have a devices directory (or doesn't require one at least), so this patch error feels wrong.

Reply:
Oh, actually the check looks correct but the error string not so much.

Reply:
The error strings are in the wrong order. fixed now